.buildkite/scripts/common.sh

```diff
@@ -69,3 +71,32 @@ with_Terraform() {
     chmod +x ${WORKSPACE}/terraform
     terraform version
 }
+
+google_cloud_auth() {
+    secretFileLocation=$(mktemp -d -p . -t "${TMP_FOLDER_TEMPLATE_BASE}.XXXXXXXXX")/google-cloud-credentials.json
+    echo "${PRIVATE_CI_GCS_CREDENTIALS_SECRET}" > ${secretFileLocation}
+    gcloud auth activate-service-account --key-file ${secretFileLocation} 2> /dev/null
+    export GOOGLE_APPLICATIONS_CREDENTIALS=${secretFileLocation}
+}
+
+upload_packages_to_gcp_backet() {
+    pattern=${1}
+    baseUrl="gs://${JOB_GCS_BUCKET}/${REPO}"
+    bucketUrlCommit="${baseUrl}"/commits/${BUILDKITE_COMMIT}
+    bucketUrlDefault="${baseUrl}"/snapshots
+
+    if [[ ${BUILDKITE_PULL_REQUEST} != "false" ]]; then
+        bucketUrlDefault="${baseUrl}"/pull-requests/pr-${BUILDKITE_COMMIT}
```

Comment:
This is generating links using the commit instead of the PR id (e.g. 2628 for this  pull request):

```
gs://fleet-ci-artifacts/fleet-server/pull-requests/pr-b10bfc08046aa6a26c05141af4a348a3306613e5
gs://fleet-ci-artifacts/fleet-server/commits/b10bfc08046aa6a26c05141af4a348a3306613e5
```

This could be changed to use the GITHUB_PR_NUMBER environment variable. However as this would match to the URL used by Jenkins, probably it could be added some sufix (e.g. PR-<id>-buildkite?)

```suggestion
        bucketUrlDefault="${baseUrl}"/pull-requests/pr-${GITHUB_PR_NUMBER}-buildkite
```

This should be updated once it is finally migrated to Buildkite from Jenkins.